cpp/src/arrow/acero/time_series_util.cc

```diff
@@ -0,0 +1,63 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/array/data.h"
+
+#include "arrow/acero/time_series_util.h"
+#include "arrow/util/logging.h"
+
+namespace arrow::acero {
+
+// normalize the value to 64-bits while preserving ordering of values
```

Comment:
```suggestion
// normalize the value to unsigned 64-bits while preserving ordering of values
```

Normalizing to 64 bits is easy.  The trick I think you are applying here is the fact that you are normalizing to an _unsigned_ value and so you have to fiddle with negative values so they compare correctly right?

Also, we can probably remove this comment since you have the same comment in the header file.